org.eclipse.xtext.xtext.ui.examples/projects/org.eclipse.xtext.example.homeautomation.tests/src/org/eclipse/xtext/example/homeautomation/tests/SerializerTest.xtend

```diff
@@ -0,0 +1,72 @@
+package org.eclipse.xtext.example.homeautomation.tests
+
+import java.io.StringWriter
+import javax.inject.Inject
+import javax.inject.Provider
+import org.eclipse.emf.common.util.URI
+import org.eclipse.xtext.example.homeautomation.ruleEngine.Device
+import org.eclipse.xtext.example.homeautomation.ruleEngine.RuleEngineFactory
+import org.eclipse.xtext.resource.SaveOptions
+import org.eclipse.xtext.resource.XtextResourceSet
+import org.eclipse.xtext.serializer.ISerializer
+import org.eclipse.xtext.testing.InjectWith
+import org.eclipse.xtext.testing.XtextRunner
+import org.eclipse.xtext.xbase.XbaseFactory
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(XtextRunner)
+@InjectWith(RuleEngineInjectorProvider)
+class SerializerTest extends Assert {
+	@Inject ISerializer serializer
+	RuleEngineFactory factory = RuleEngineFactory.eINSTANCE
```

Comment:
Basically I thought so to make the interaction with the serializer more explicit. Do you think using it as extension would be better?